student/src/main/java/tds/student/services/remote/RemoteItemScoringService.java

```diff
@@ -0,0 +1,108 @@
+import tds.student.sql.repository.remote.ItemScoringRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * This IItemScoringService implementation forwards scoring requests to the ItemScoringRepository.
+ */
+@Service("integrationItemScoringService")
+@Scope("prototype")
+public class RemoteItemScoringService implements IItemScoringService {
+
+    private final IItemScoringService legacyItemScoringService;
+    private final boolean remoteExamCallsEnabled;
+    private final boolean legacyCallsEnabled;
+    private final ItemScoringRepository repository;
+
+
+    @Autowired
+    public RemoteItemScoringService(
+        @Qualifier("legacyItemScoringService") final IItemScoringService legacyItemScoringService,
+        @Value("${tds.exam.remote.enabled}") final Boolean remoteExamCallsEnabled,
+        @Value("${tds.exam.legacy.enabled}") final Boolean legacyCallsEnabled,
+        final ItemScoringRepository repository) {
+
+        if (!remoteExamCallsEnabled && !legacyCallsEnabled) {
+            throw new IllegalStateException("Remote and legacy calls are both disabled.  Please check progman configuration");
+        }
+
+        this.legacyItemScoringService = legacyItemScoringService;
+        this.remoteExamCallsEnabled = remoteExamCallsEnabled;
+        this.legacyCallsEnabled = legacyCallsEnabled;
+        this.repository = repository;
+    }
+
+    @Override
+    public List<ItemResponseUpdateStatus> updateResponses(final OpportunityInstance oppInstance,
+                                                          final List<ItemResponseUpdate> responsesUpdated,
+                                                          final Float pageDuration) throws ReturnStatusException {
+        List<ItemResponseUpdateStatus> responseUpdateStatuses = new ArrayList<>();
+
+        if (legacyCallsEnabled) {
+            responseUpdateStatuses = legacyItemScoringService.updateResponses(oppInstance, responsesUpdated, pageDuration);
+        }
+
+        if (!remoteExamCallsEnabled) {
+            return responseUpdateStatuses;
+        }
+
+        responseUpdateStatuses = repository.updateResponses(
+            oppInstance.getExamId(),
+            oppInstance.getSessionKey(),
+            oppInstance.getExamBrowserKey(),
+            oppInstance.getExamClientName(),
+            pageDuration,
+            responsesUpdated);
+
+        return responseUpdateStatuses;
+    }
+
+    @Override
+    public ItemScore checkScoreability(final IItemResponseScorable responseScorable, final IITSDocument itsDoc) throws ReturnStatusException {
+        if (legacyCallsEnabled) {
+            return legacyItemScoringService.checkScoreability(responseScorable, itsDoc);
+        }
+
+        //This method is only called via a legacy service self-reference and via the unused TimerScoringService
+        throw new UnsupportedOperationException("No remote implementation of checkScoreability");
+    }
```

Comment:
Some of these methods don't have a remote implementation because they're only called from inside the legacy implementation OR unused "Timer" classes.